### DISCUSSION TITLE
Export keyframes property

### DISCUSSION THREAD

**Discussion by KarelVerschraegen** [ben-rogerson/twin.macro#92] [Ideas]
Since both emotion and styled-components have a keyframes helper, would it be possible to export keyframes from within twin itself?

This would be in line with the `styled` export from twin.

**Current behaviour**
```
import { keyframes } from 'styled-components';
import tw, { styled } from 'twin.macro';

const component = styled.div`
    ${tw`mt-4`}
`

const animation = keyframes`
    ...
`;
```

**Expected behaviour**
```
import tw, { keyframes, styled } from 'twin.macro';

const component = styled.div`
    ${tw`mt-4`}
`

const animation = keyframes`
    ...
`;
```

If pointed in the right direction, I'd gladly (try to) implement this myself!

**Comment by ben-rogerson** [comment]
I think that adding the keyframe import would make a good feature.

If you'd like to have a crack at it, here are some tasks I can think of:

- [ ] Add [keyframe import presets](https://github.com/ben-rogerson/twin.macro/blob/master/src/config/userPresets.js) for emotion + styled-components
- [ ] Add the keyframe function replacement with babel-plugin-macros - you could copy and adapt the existing [styled import code](https://github.com/ben-rogerson/twin.macro/blob/master/src/macro.js#L90)
- [ ] Add 'keyframes' to both the import validator and list of suggestions
- [ ] Test to make sure it doesn't unnecessarily slow down the compilation speed
- [ ] Add a [keyframe import fixture test](https://github.com/ben-rogerson/twin.macro/blob/master/__fixtures__/!imports.js)
- [ ] Update the docs with the new feature
- [ ] Update the [typescript section](https://github.com/ben-rogerson/twin.macro/blob/master/docs/typescript.md) with new instructions

I can help out if you need a hand with anything.

### Resources
[Keyframes - emotion](https://emotion.sh/docs/keyframes)
[Keyframes - styled-components](https://styled-components.com/docs/api#keyframes)

**Comment by fvanwijk** [comment]
Possibly related to this feature request, I'd like to get rid of the global PostCSS build, only include `tailwindcss/dist/base.min.css` and let twin do the rest.
So far the results are good, but the keyframes aren't part of the base styles.
Thus I would expect twin to include them when I use one of the `animate-xxx` utils.

[Animation core plugin](https://github.com/tailwindlabs/tailwindcss/blob/master/src/plugins/animation.js)

**Comment by ben-rogerson** [comment]
Thanks for bringing this up @fvanwijk - I'm keen to hear your thoughts on a good solution over in #127.

**Comment by dominik-sfl** [comment]
are keyframes imports added yet?
